### PR TITLE
MAINT: Correct code producing warnings

### PR DESCRIPTION
**Pull request by bashtage**
Cast to avoid warnings
Correct function

### CODE REVIEW

**Comment by bashtage** [comment]
These produce warnings when building using clang-cl on Windows. Two are just casts. The other appears to be the wrong function for the input (labs -> llabs).

**Comment by bashtage** [comment]
Failure is due to coverage, not being wrong.

File: numpy/core/src/multiarray/ctors.c [hunk 1]
```diff
@@ -2907,7 +2907,7 @@ _arange_safe_ceil_to_intp(double value)
             "arange: cannot compute length");
         return -1;
     }
-    if (!(NPY_MIN_INTP <= ivalue && ivalue <= NPY_MAX_INTP)) {
+    if (!(NPY_MIN_INTP <= ivalue && ivalue <= (double)NPY_MAX_INTP)) {
```
Comment (mattip) [inline numpy/core/src/multiarray/ctors.c:2910]: This is strange: shouldn't you need to cast both `NPY_MIN_INTP` and `NPY_MAX_INTP`?

**Comment by bashtage** [comment]
It doesn't complain about NPY_MIN_INTP, just NPY_MAX_INTP. Could the double representation of NPY_MIN_INTP is exact?

```
2021-02-15T14:35:31.0278557Z numpy\core\src\multiarray\ctors.c(2910,47): warning: implicit conversion from 'long long' to 'double' changes value from 9223372036854775807 to 9223372036854775808 [-Wimplicit-const-int-float-conversion]
2021-02-15T14:35:31.0280755Z     if (!(NPY_MIN_INTP <= ivalue && ivalue <= NPY_MAX_INTP)) {
2021-02-15T14:35:31.0281923Z                                            ~~ ^~~~~~~~~~~~
2021-02-15T14:35:31.0283215Z numpy\core\include\numpy/npy_common.h(297,30): note: expanded from macro 'NPY_MAX_INTP'
2021-02-15T14:35:31.0284588Z         #define NPY_MAX_INTP NPY_MAX_LONGLONG
2021-02-15T14:35:31.0285689Z                              ^~~~~~~~~~~~~~~~
2021-02-15T14:35:31.0286967Z numpy\core\include\numpy/npy_common.h(637,28): note: expanded from macro 'NPY_MAX_LONGLONG'
2021-02-15T14:35:31.0288182Z #  define NPY_MAX_LONGLONG NPY_MAX_INT64
2021-02-15T14:35:31.0289226Z                            ^~~~~~~~~~~~~
2021-02-15T14:35:31.0290720Z numpy\core\include\numpy/npy_common.h(449,23): note: expanded from macro 'NPY_MAX_INT64'
2021-02-15T14:35:31.0291996Z #define NPY_MAX_INT64 NPY_LONGLONG_SUFFIX(9223372036854775807)
2021-02-15T14:35:31.0293243Z                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2021-02-15T14:35:31.0294506Z numpy\core\include\numpy/npy_common.h(344,39): note: expanded from macro 'NPY_LONGLONG_SUFFIX'
2021-02-15T14:35:31.0295780Z #    define NPY_LONGLONG_SUFFIX(x)   (x##i64)
2021-02-15T14:35:31.0296924Z                                       ^~~~~~
2021-02-15T14:35:31.0297733Z <scratch space>(126,1): note: expanded from here
2021-02-15T14:35:31.0298576Z 9223372036854775807i64
2021-02-15T14:35:31.0299225Z ^~~~~~~~~~~~~~~~~~~~~~
```

**Comment by mattip** [comment]
I think even if it does not currently emit a warning it would be better to pre-emptively add a cast to `NPY_MIN_INTP` as well.

**Comment by bashtage** [comment]
Easy enough to add.

**Review by mattip — APPROVED** [review]
LGTM once CI passes

**Comment by bashtage** [comment]
@mattip all green except coverage delta which is unavoidable.

**Comment by mattip** [comment]
Thanks @bashtage

**Pull request merged**